### ISSUE TITLE
undefined method `name' in Style/MethodCallParentheses

### ISSUE THREAD

**Issue by kollee** [bbatsov/rubocop#3307]
In our code we had something like

```ruby
dates = {}
dates['result'] = Time.now
dates['date'] = Time.now - (48 * 3600)
dates['result'] += (Time.now() - dates['date'])
```

Notice the unnecessary parentheses for ``Time.now`` in line 4.  Running ``rubocop`` on this will produce a ``undefined method`` error while CRuby executes it just fine. After removing the parentheses the error is gone. If you rewrite the last line into

```ruby
dates['result'] = dates['result'] + (Time.now() - dates['date'])
```

``rubocop`` works fine complaining about ``Style/MethodCallParentheses: Do not use parentheses for method calls with no arguments.``

--------

## Expected behavior
```
$ rubocop -d example.rb
For /project: configuration from /project/.rubocop.yml
Default configuration from project/gems/rubocop-0.41.2/config/default.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/enabled.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/disabled.yml
Inspecting 1 file
Scanning /project/test.rb
For /project: configuration from /project/gems/rubocop-0.41.2/config/default.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/enabled.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/disabled.yml
C

Offenses:

/project/example.rb:4:28: C: Style/MethodCallParentheses: Do not use parentheses for method calls with no arguments.
dates['result'] += (Time.now() - dates['date'])
                            ^

1 file inspected, 1 offense detected
Finished in 0.07414920500013977 seconds
```

## Actual behavior
```
$ rubocop -d example.rb
An error occurred while Style/MethodCallParentheses cop was inspecting /project/example.rb.

1 error occurred:
An error occurred while Style/MethodCallParentheses cop was inspecting /project/example.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.41.2 (using Parser 2.3.1.2, running on ruby 2.2.4 x86_64-linux)
For /project: configuration from /project/.rubocop.yml
Default configuration from /project/gems/rubocop-0.41.2/config/default.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/enabled.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/disabled.yml
Inspecting 1 file
Scanning /project/example.rb
For /project: configuration from /project/gems/rubocop-0.41.2/config/default.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/enabled.yml
Inheriting configuration from /project/gems/rubocop-0.41.2/config/disabled.yml
undefined method `name' for #<Parser::Source::Map::Send:0x005633c2489df0>
/project/gems/rubocop-0.41.2/lib/rubocop/cop/style/method_call_parentheses.rb:57:in `block in same_name_assignment?'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node.rb:548:in `visit_ancestors_with_types'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node.rb:147:in `each_ancestor'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/style/method_call_parentheses.rb:39:in `each'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/style/method_call_parentheses.rb:39:in `any?'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/style/method_call_parentheses.rb:39:in `same_name_assignment?'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/style/method_call_parentheses.rb:20:in `on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:42:in `block (2 levels) in on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:97:in `with_cop_error_handling'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:41:in `block in on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:40:in `each'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:40:in `on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:102:in `block in on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:100:in `each'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:100:in `each_with_index'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:100:in `on_send'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:46:in `on_send'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:46:in `on_begin'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:114:in `on_op_asgn'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:46:in `on_op_asgn'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:46:in `on_begin'
/project/gems/rubocop-0.41.2/lib/rubocop/ast_node/traversal.rb:13:in `walk'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/commissioner.rb:59:in `investigate'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/team.rb:121:in `investigate'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/team.rb:109:in `offenses'
/project/gems/rubocop-0.41.2/lib/rubocop/cop/team.rb:52:in `inspect_file'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:212:in `inspect_file'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:182:in `block in do_inspection_loop'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:172:in `loop'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:172:in `do_inspection_loop'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:87:in `process_file'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:59:in `block in inspect_files'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:57:in `each'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:57:in `inspect_files'
/project/gems/rubocop-0.41.2/lib/rubocop/runner.rb:35:in `run'
/project/gems/rubocop-0.41.2/lib/rubocop/cli.rb:72:in `execute_runner'
/project/gems/rubocop-0.41.2/lib/rubocop/cli.rb:28:in `run'
/project/gems/rubocop-0.41.2/bin/rubocop:14:in `block in <top (required)>'
/home/kollee/.rbenv/versions/2.2.4/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/proejct/gems/rubocop-0.41.2/bin/rubocop:13:in `<top (required)>'
/project/./gems/bin/rubocop:23:in `load'
/project/./gems/bin/rubocop:23:in `<main>'
.

1 file inspected, no offenses detected
Finished in 0.10808629900020605 seconds
```

## Steps to reproduce the problem

Use the provided example.

## RuboCop version
```
$ rubocop -V
0.41.2 (using Parser 2.3.1.2, running on ruby 2.2.4 x86_64-linux)
```
